### DISCUSSION TITLE
Manually placing items in a ScrollArea

### DISCUSSION THREAD

**Discussion by edko99** [emilk/egui#449] [Q&A]
I would like to display a list that could potentially grow to many thousands of items. Instead of adding all the items to the UI, I would rather just place the ones that will be visible in that frame, manually inside a ScrollArea. Is there a way to do this?

**Comment by emilk** [comment]
Very good question!

Manual layout is certainly possible, with either `ui.put` or adding a sub-ui that covers the visible part with `ui.child_ui`.

I think the one piece missing is reading how far scrolled down we are. That may require some design work.

I've opened an issue on this at https://github.com/emilk/egui/issues/451
